crates/estree/src/util.rs

```diff
@@ -0,0 +1,97 @@
+use serde_json::Value;
+use serde::ser::*;
+
+pub enum Container {
+    NothingMeaningful,
+    FunctionDeclaration,
+    FunctionExpression,
+}
+
+/// A container used to represent some piece of data being
+/// serialized. Future versions may add options to the
+/// serialization format.
+pub struct Serialization<'a, T> where T: 'a {
```

Comment:
TL;DR - I'm suggesting to just provide a trait that would allow to convert any Easter node to serde_json::Map, and then we don't need to care about serialization config in the Esprit itself, but instead let consumer decide what to do with that JSON tree further.